packages/snaps-utils/src/interface.ts

```diff
@@ -0,0 +1,9 @@
+import type { Infer } from 'superstruct';
+import { nullable, record, string, union } from 'superstruct';
+
+export const InterfaceStateStruct = record(
+  string(),
+  union([record(string(), nullable(string())), nullable(string())]),
```

Comment:
```suggestion
  record(string(), nullable(string())),
```
If we assume every input to have a form then we can simplify